src/Microsoft.ComponentDetection.Common/DockerReference/DigestUtility.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+
+namespace Microsoft.ComponentDetection.Common
+{
+    public class DigestUtility
+    {
+        private static readonly Dictionary<string, int> AlgorithmsSizes = new Dictionary<string, int>()
+        {
+            { "sha256", 32 },
+            { "sha384", 48 },
+            { "sha512", 64 },
+        };
+
+        public static bool CheckDigest(string digest, bool throwError = true)
```

Comment:
I think the C# equivalent would be to have `TryCheckDigest(string digest)` which returns `bool`, and `CheckDigest(string digest)` which can throw.